### PR TITLE
Cli building

### PR DESCRIPTION
**Pull request by jarsba**
- Add segmentation scripts
- Fix some training and prediction todo

### CODE REVIEW

File: preprocessing/preprocess_split_data.py [hunk 1]
```diff
@@ -7,61 +7,68 @@
...
 import cv2
 
+def preprocess_split_data():
+    # %%
 
-# %%
+    pd.set_option('display.max_rows', 100)
+    pd.set_option('display.max_columns', 30)
+    pd.options.display.max_colwidth = 100
+    # %%
```
Comment (Jakoviz) [inline preprocessing/preprocess_split_data.py:16]: These are redundant I suppose.

File: preprocessing/preprocess_split_data.py [hunk 2]
```diff
@@ -7,61 +7,68 @@
...
 
-plant_df_split = pd.read_csv(f'{DATA_FOLDER_PATH}/plant_data_split.csv')
-# %%
+    growth_plant_df_split = pd.read_csv(f'{DATA_FOLDER_PATH}/growth_chamber_plant_data_split.csv')
+    # %%
 
-growth_plant_df_split = pd.read_csv(f'{DATA_FOLDER_PATH}/growth_chamber_plant_data_split.csv')
-# %%
+    plant_df_split = plant_df_split[["Trial", "Dataset", "Genotype", "Condition", "Original image path", "Masked image path", "Split masked image path"]]
+    plant_df_split
```
Comment (Jakoviz) [inline preprocessing/preprocess_split_data.py:33]: Redundant

File: preprocessing/preprocess_split_data.py [hunk 3]
```diff
@@ -7,61 +7,68 @@
...
+    # %%
 
-plant_df_split = plant_df_split[["Trial", "Dataset", "Genotype", "Condition", "Original image path", "Masked image path", "Split masked image path"]]
-plant_df_split
-# %%
+    growth_plant_df_split = growth_plant_df_split[["Trial", "Dataset", "Genotype", "Condition", "Original image path", "Masked image path", "Split masked image path"]]
+    growth_plant_df_split
```
Comment (Jakoviz) [inline preprocessing/preprocess_split_data.py:37]: Redundant

File: preprocessing/preprocess_split_data.py [hunk 4]
```diff
@@ -7,61 +7,68 @@
...
+    # %%
 
-plant_df_split.dropna(inplace=True)
-# %%
+    plant_df_split_master = pd.concat([plant_df_split, growth_plant_df_split])
+    # %%
 
-plant_df_split_master = pd.concat([plant_df_split, growth_plant_df_split])
-# %%
+    plant_df_split_master
```
Comment (Jakoviz) [inline preprocessing/preprocess_split_data.py:51]: Redundant

File: preprocessing/preprocess_split_data.py [hunk 5]
```diff
@@ -7,61 +7,68 @@
...
+    # %%
 
-plant_df_split_master
-# %%
+    plant_df_split_master['Label Category'] = pd.Categorical(plant_df_split_master['Condition'])
+    plant_df_split_master['Label Category'] = pd.Categorical(plant_df_split_master['Condition'])
+    plant_df_split_master['Label'] = plant_df_split_master['Label Category'].cat.codes
+    plant_df_split_master
```
Comment (Jakoviz) [inline preprocessing/preprocess_split_data.py:57]: Redundant

**Review by Jakoviz — APPROVED** [review]
Very nice. Just some redundant lines of code that I pointed out.

File: utils/model_utils.py [hunk 6]
```diff
@@ -30,6 +28,18 @@
...
+	"resnet18": {
+		"image_size": (256, 256)
+	},
+	"inception_v3": {
+		"image_size": (299, 299)
+	},
+	"vision_transformer": {
+		"image_size": (256, 256)
+	}
+}
```
Comment (Redande) [inline utils/model_utils.py:41]:
```suggestion
	"vision_transformer": {
		"image_size": (256, 256)
	},
	"bag_of_words": {
		"image_size": (256, 256)
	}
}
```

Comment (Redande) [reply utils/model_utils.py:41]: Bag of words doesn't really need image resizing, but to conform to other classifiers and not overcomplicate data augmentation in train.py by having multiple augmentations depending on model architecture, we could add this here

**Review by Redande — APPROVED** [review]

**Pull request merged**